### ISSUE TITLE
[PORT] Fixes Unauthorized error when calling ContinueConversation

### ISSUE THREAD

**Issue by tomlm** [microsoft/botbuilder-js#2621]
> Port this change from botbuilder-dotnet/master branch:
https://github.com/microsoft/botbuilder-dotnet/pull/4348

Fixes #4347

Fixes Unauthorized error when ContinueConversation is called on a bot app service instance that didn't originate the skill request.
- Updated code in BotFrameworkAdapter.ContinueConversation() so it always adds the activity.ServiceUrl to TrustedHostNames if the request has been authorized and it has an app ID (this will work for skills and other proactive messages)
- Made TrustedHostNames in AppCredentials internal and added InternalsVisibleTo directive to Microsoft.BotConnector so we can access it and assert the state of the state in tests.
- Cleaned up some async warnings in BotFrameworkAdapterTests
- Added additional logging to AppCredentials to help troubleshoot this issue in the future.

# Changed projects
* Microsoft.Bot.Builder
* Microsoft.Bot.Connector
* Microsoft.Bot.Builder.Tests

**Comment by gabog** [comment]
Not needed in JS

**Comment by guy-microsoft** [comment]
I experience this issue too - I want to use BotFrameworkAdapter.ContinueConversation(...) from the skill side, for sending conversation time-out message to the root bot after a period of time of inactivity, and I also get Unauthorized error.
Is there a solution for that in JS? If yes, what is the solution?

**Comment by gabog** [comment]
@guy-microsoft, if you are trying to send a proactive message, see if any of these two articles help:
https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0&tabs=javascript
https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0&tabs=javascript#avoiding-401-unauthorized-errors

**Comment by guy-microsoft** [comment]
@gabog I have already followed the steps in this docs, and added the url of the root boot to the list of trusted host names, but it didn't help:
I managed to run a Skill bot that doesn't send proactive messages, and I managed to run a non-Skill bot that sends proactive messages to the user. However, when I try to continueConversation(...) with the adapter of a skill bot, and to send a proactive message using the context that is created, I get Unauthorized error.
Are you familiar with that issue?
